File: week 1/prog4.c
void moveZeroes(int arr[], int n) {
    int count = 0; // count of non-zero elements

    // Move all non-zero elements to front
    for(int i = 0; i < n; i++) {
        if(arr[i] != 0) {
            arr[count] = arr[i];
            count++;
        }
    }

    // Fill remaining positions with zero
    while(count < n) {
        arr[count] = 0;
        count++;
    }
}

File: week 1/python programming/prog4.py
arr = [0, 1, 0, 3, 12]
count = 0

for i in range(len(arr)):
    if arr[i] != 0:
        arr[count] = arr[i]
        count += 1

while count < len(arr):
    arr[count] = 0
    count += 1

print(arr)
